src/Appwrite/Databases/TransactionState.php

```
@@ -0,0 +1,640 @@
+        string $collectionId,
+        Document $updateData,
+        array $queries,
+        array &$state
+    ): void {
+        if (!isset($state[$collectionId])) {
+            return;
+        }
+
+        $filters = $this->extractFilters($queries);
+
+        foreach ($state[$collectionId] as $docId => $doc) {
```

PHPMD (2.15.0)

Avoid unused local variables such as '$docId'. (undefined)

(UnusedLocalVariable)